rubicon/objc/runtime.py

```diff
@@ -2097,14 +2105,21 @@ def __init__(self, func, restype=NOTHING, *arg_types):
 
         self.literal = BlockLiteral()
         self.literal.isa = addressof(_NSConcreteGlobalBlock)
-        self.literal.flags = BlockConsts.HAS_STRET | BlockConsts.HAS_SIGNATURE
+        self.literal.flags = BlockConsts.HAS_STRET | BlockConsts.HAS_SIGNATURE | BlockConsts.HAS_COPY_DISPOSE
         self.literal.reserved = 0
         self.cfunc = self.cfunc_type(self.wrapper)
         self.literal.invoke = cast(self.cfunc, c_void_p)
         self.descriptor = BlockDescriptor()
         self.descriptor.reserved = 0
         self.descriptor.size = sizeof(BlockLiteral)
 
+        self.copy_helper_type = CFUNCTYPE(c_void_p, c_void_p, c_void_p)
+        self.dispose_helper_type = CFUNCTYPE(c_void_p, c_void_p)
+        self.copy_helper = self.copy_helper_type(self.copyHelper)
+        self.dispose_helper = self.dispose_helper_type(self.disposeHelper)
+        self.descriptor.copy_helper = cast(self.copy_helper, c_void_p)
+        self.descriptor.dispose_helper = cast(self.dispose_helper, c_void_p)
```

Comment:
See above, these casts can be removed if the struct field types are set correctly.